server/src/main/kotlin/org/itmodreamteam/myrest/server/service/sms/RealSmsService.kt

```diff
@@ -0,0 +1,93 @@
+package org.itmodreamteam.myrest.server.service.sms
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import org.itmodreamteam.myrest.server.error.UserException
+import org.slf4j.LoggerFactory
+import org.springframework.boot.web.client.RestTemplateBuilder
+import org.springframework.context.annotation.Primary
+import org.springframework.context.annotation.Profile
+import org.springframework.data.domain.PageRequest
+import org.springframework.data.domain.Sort
+import org.springframework.stereotype.Service
+import org.springframework.web.client.postForEntity
+import java.nio.charset.Charset
+import java.time.*
+
+@Service
+@Primary
+@Profile("prod")
+class RealSmsService(private var smsMessageRepository: SmsMessageRepository) : SmsService {
+
+    companion object {
+        private val MIN_INTERVAL_BETWEEN_MESSAGES = Duration.ofMinutes(5)
+    }
+
+    private val log = LoggerFactory.getLogger(javaClass)
+
+    private val restTemplate = RestTemplateBuilder()
+        .rootUri("https://rest-api.d7networks.com/secure")
```

Comment:
Is it okay to hardcode the API address?

Reply:
I don't see anything bad in it. Though, it may be better to configure this `restTemplate` in a more flexible way (Spring Bean) for more granular testing.